rir/src/interpreter/interp.cpp

```diff
@@ -2130,18 +2130,9 @@ SEXP evalRirCode(Code* c, InterpreterInstance* ctx, SEXP env,
             if (fun->invocationCount() % PIR_WARMUP == 0) {
                 Assumptions assumptions =
                     addDynamicAssumptionsForOneTarget(call, fun->signature());
-                if (assumptions != fun->signature().assumptions) {
+                if (assumptions != fun->signature().assumptions)
                     // We have more assumptions available, let's recompile
                     dispatchFail = true;
-
```

Comment:
Sorry for the late comment. But I do not understand this change. This does not seem like it was a bug for me but a design decision. What is the precise reason we are taking that decision down?